chain/transaction.go

```diff
@@ -359,6 +362,92 @@ func (t *Transaction) Marshal(p *codec.Packer) error {
 	return t.marshal(p)
 }
 
+func (t *Transaction) MarshalJSON() ([]byte, error) {
+	type txJSON struct {
+		ID      ids.ID `json:"id"`
+		Actions []byte `json:"actions"`
+		Auth    []byte `json:"auth"`
+	}
+
+	actionsPacker := codec.NewWriter(0, consts.NetworkSizeLimit)
+	err := t.Actions.marshalInto(actionsPacker)
+	if err != nil {
+		return nil, err
+	}
+	if err := actionsPacker.Err(); err != nil {
+		return nil, err
+	}
+	authPacker := codec.NewWriter(0, consts.NetworkSizeLimit)
+	authPacker.PackByte(t.Auth.GetTypeID())
+	t.Auth.Marshal(authPacker)
+	if err := authPacker.Err(); err != nil {
+		return nil, err
+	}
+
+	hexActions, err := codec.Bytes(actionsPacker.Bytes()).MarshalText()
+	if err != nil {
+		return nil, err
+	}
+	hexAuth, err := codec.Bytes(authPacker.Bytes()).MarshalText()
+	if err != nil {
+		return nil, err
+	}
+
+	return json.Marshal(txJSON{
+		ID:      t.ID(),
+		Actions: hexActions,
+		Auth:    hexAuth,
+	})
+}
+
+func (t *Transaction) UnmarshalJSON(data []byte, parser Parser) error {
+	type txJSON struct {
+		ID      ids.ID `json:"id"`
+		Actions []byte `json:"actions"`
+		Auth    []byte `json:"auth"`
+	}
+
+	var tx txJSON
+	err := json.Unmarshal(data, &tx)
+	if err != nil {
+		return err
+	}
+
+	var actionsBytes, authBytes codec.Bytes
+	err = actionsBytes.UnmarshalText(tx.Actions)
+	if err != nil {
+		return fmt.Errorf("%w: cannot unmarshal actions text", err)
+	}
+	err = authBytes.UnmarshalText(tx.Auth)
+	if err != nil {
+		return fmt.Errorf("%w: cannot unmarshal auth text", err)
+	}
+
```

Comment:
Same comment for removing `UnmarshalText` in favor of using `codec.Bytes` in the `txJSON` type